python/lsst/afw/typehandling/_SimpleGenericMap.py

```diff
@@ -19,10 +19,48 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-__all__ = []
+__all__ = ["SimpleGenericMap"]
 
 from lsst.utils import continueClass
 from ._typehandling import SimpleGenericMapS
+from ._GenericMap import MutableGenericMap, AutoKeyMeta
+
+
+class SimpleGenericMap(MutableGenericMap, metaclass=AutoKeyMeta):
+    """A `dict`-like `~collections.abc.MutableMapping` for use when sharing a
+    map between C++ and Python.
+
+    For compatibility with C++, ``SimpleGenericMap`` has the following
+    restrictions:
+
+        - all keys must be of the same type
+        - values must be built-in types or subclasses of
+          `lsst.afw.typehandling.Storable`. Almost any user-defined class in
+          C++ or Python can have `~lsst.afw.typehandling.Storable` as a mixin.
+
+    As a safety precaution, `~lsst.afw.typehandling.Storable` objects that are
+    added from C++ may be copied when you retrieve them from Python, making it
+    impossible to modify them in-place. This issue does not affect objects that
+    are added from Python, or objects that are always passed by
+    :cpp:class:`shared_ptr` in C++.
```

Comment:
It's too bad that the above docstring has to be copied into each of the different `*GenericMap`s. I don't think there's a way around it though.

Reply:
Well, in pybind11 I could factor the common text into its own function. Would it make sense to do the same here (assigning to `__doc__`)?

Reply:
All of the various alternate approaches I've seen feel not great and/or hacky to me, so I think we're stuck with it. Unless @jonathansick has any better ideas?